src/adapters/acala.ts

```diff
@@ -0,0 +1,231 @@
+import { Wallet } from "@acala-network/sdk/wallet";
+import { AnyApi, FixedPointNumber } from "@acala-network/sdk-core";
+import {
+  catchError,
+  combineLatest,
+  firstValueFrom,
+  map,
+  Observable,
+  of,
+} from "rxjs";
+
+import "@acala-network/types/argument/api-tx";
+import { ApiRx } from "@polkadot/api";
+import { SubmittableExtrinsic } from "@polkadot/api/types";
+import { ISubmittableResult } from "@polkadot/types/types";
+
+import { BaseCrossChainAdapter } from "../base-chain-adapter";
+import { ChainName, chains } from "../configs";
+import { ApiNotFound } from "../errors";
+import {
+  BalanceData,
+  BasicToken,
+  CrossChainRouterConfigs,
+  CrossChainTransferParams,
+} from "../types";
+
+const ACALA_DEST_WEIGHT = "5000000000";
+
+// export const acalaRoutersConfig: Omit<CrossChainRouterConfigs, "from">[] = [
+//   {
+//     to: "interlay",
+//     token: "INTR",
+//     xcm: {
+//       fee: { token: "INTR", amount: "21787589" },
+//       weightLimit: ACALA_DEST_WEIGHT,
+//     },
+//   },
+//   {
+//     to: "interlay",
+//     token: "IBTC",
+//     xcm: {
+//       fee: { token: "IBTC", amount: "72" },
+//       weightLimit: ACALA_DEST_WEIGHT,
+//     },
+//   },
+// ];
+
+export const karuraRoutersConfig: Omit<CrossChainRouterConfigs, "from">[] = [
+  {
+    to: "kintsugi",
+    token: "KINT",
+    xcm: {
+      fee: { token: "KINT", amount: "300000000" },
+      weightLimit: ACALA_DEST_WEIGHT,
+    },
+  },
+  {
+    to: "kintsugi",
+    token: "KBTC",
+    xcm: {
+      fee: { token: "KBTC", amount: "200" },
+      weightLimit: ACALA_DEST_WEIGHT,
+    },
+  },
+  //   {
+  //     to: "kintsugi",
+  //     token: "LKSM",
+  //     xcm: {
+  //       fee: { token: "LKSM", amount: "186480000" },
+  //       weightLimit: ACALA_DEST_WEIGHT,
+  //     },
+  //   },
+];
+
+// export const acalaTokensConfig: Record<string, BasicToken> = {
+//   INTR: { name: "INTR", symbol: "INTR", decimals: 10, ed: "1000000000" },
+//   IBTC: { name: "IBTC", symbol: "IBTC", decimals: 8, ed: "100" },
+// };
+
+export const karuraTokensConfig: Record<string, BasicToken> = {
+  //   LKSM: { name: "LKSM", symbol: "LKSM", decimals: 12, ed: "500000000" },
+  KINT: { name: "KINT", symbol: "KINT", decimals: 12, ed: "133330000" },
```

Comment:
Can you add sources on the ED?

Reply:
Good idea. 

So far, unless a comment says otherwise (ie. I could look it up on-chain), the source is "what was used in polkawallet-io/bridge" :)

Reply:
Done.